boards/arduino-atmega-common/include/periph_conf.h

```diff
@@ -142,6 +142,41 @@ extern "C" {
 #endif
 /** @} */
 
+/**
+ * @brief ADC configuration
+ *
+ * @{
+ */
+#ifdef CPU_ATMEGA328P
+#define ADC_NUMOF       (6)
```

Comment:
au contraire, Arduino Duemilanove has only 6 analog in pins accessible, though there may be 8 internally. See [here](https://www.arduino.cc/en/Main/ArduinoBoardDuemilanove), picture and reference table.